client/src/Components/Layout/Login/Login.jsx

```diff
@@ -0,0 +1,331 @@
+import React, { Component } from 'react';
+import queryString from 'query-string';
+import { Link, Redirect } from 'react-router-dom';
+import {
+  Grid,
+  Box,
+  Typography,
+  Button,
+  Paper,
+  TextField,
+  FormHelperText,
+  FormControl,
+  Input,
+  InputLabel,
+  IconButton,
+  InputAdornment,
+} from '@material-ui/core';
+import {
+  Person as PersonIcon,
+  Lock as LockIcon,
+  Visibility,
+  VisibilityOff,
+} from '@material-ui/icons';
+import Cookies from 'js-cookie';
+import { withStyles } from '@material-ui/core/styles';
+
+import LoaderProgress from '../../Common/LoaderProgress';
+
+import axios from 'axios';
+import LoginStyles from './LoginStyle';
+
+class Login extends Component {
+  state = {
+    logindata: { email: null, password: null },
+    isLoading: false,
+    displayBlock: true,
+    showPassword: false,
+    email: {
+      Message: '',
+      isValid: true,
+    },
+    password: {
+      Message: '',
+      isValid: true,
+      minLength: 3,
+    },
+    redirect: false,
+    ReturnUrlText: '/',
+    SignupLink: '/user/Signup/',
+  };
+
+  componentDidMount() {
+    const AuthToken = Cookies.get('AuthToken');
+
+    console.log(AuthToken);
+    console.log(!AuthToken);
+    const qstring = queryString.parse(this.props.history.location.search);
+    if (qstring.ReturnUrl) {
+      this.setState({
+        ReturnUrlText: qstring.ReturnUrl,
+        SignupLink: `/user/Signup/?ReturnUrl=${qstring.ReturnUrl}`,
+      });
+    } else {
+      this.setState({
+        // if the user log in we will redirect to Home
+        redirect: !AuthToken ? false : true,
+      });
+    }
+  }
+
+  renderAction = () => {
+    if (this.state.redirect) {
+      return <Redirect to={this.state.ReturnUrlText} />;
+    }
+  };
+
+  texthandler = (e) => {
```

Comment:
You could make this name a bit more descriptive, something like `handleTextInput`

Reply:
done